go/test/endtoend/reparent/newfeaturetest/reparent_test.go

```diff
@@ -55,3 +55,43 @@ func TestRecoverWithMultipleVttabletFailures(t *testing.T) {
 	newPrimary := utils.GetNewPrimary(t, clusterInstance)
 	utils.ConfirmReplication(t, newPrimary, []*cluster.Vttablet{tablets[2], tablets[3]})
 }
+
+// TetsSingeReplicaERS tests that ERS works even when there is only 1 tablet left
+// as long the durability policy allows this failure. Moreover, this also tests that the
+// replica is one such that it was a primary itself before. This way its executed gtid set
+// will have atleast 2 tablets in it. We want to make sure this tablet is not marked as errant
+// and ERS succeeds.
+func TestSingleReplicaERS(t *testing.T) {
+	// Set up a cluster with none durability policy
+	defer cluster.PanicHandler(t)
+	clusterInstance := utils.SetupReparentCluster(t, "none")
+	defer utils.TeardownCluster(clusterInstance)
+	tablets := clusterInstance.Keyspaces[0].Shards[0].Vttablets
+	// Confirm that the replication is setup correctly in the beginning
```

Comment:
Probably worth a comment that tablets[0] is the PRIMARY at this point, although the code shows it.

Reply:
Also a note that the reparent endtoend tests — and endtoend tests in general — will become problematic at some point if they are always growing. Would be great, IMO, if we could find way to rely more on unit tests.  Is the unit test not enough in this case?

Reply:
The unit test passes and the end to end test fails, so I'll say no. Maybe I can add a couple more unit tests. In general, I merge some of these tests after 1 release. i.e. we already have tests that test reparenting, I can do this as part of a different test, in the ers package, the only reason I don't is because those tests also run in upgrade-downgrade and this one would fail there.